File: ws/collahuazo/u2/ws13GradesStudentAppCLI/JavaScript/controller/jsonManager.js
const fs = require("fs");
const path = require("path");
const Student = require("../model/student");
const Grade = require("../model/grade");

const ROOT_PATH = path.resolve(__dirname, "..");
const STUDENTS_FILE = path.join(ROOT_PATH, "students.json");
const GRADES_FILE = path.join(ROOT_PATH, "grades.json");

class JsonManager {
  static saveStudents(students) {
    fs.writeFileSync(STUDENTS_FILE, JSON.stringify(students.map((student) => student.toJson()), null, 2), "utf8");
  }

  static loadStudents() {
    if (!fs.existsSync(STUDENTS_FILE)) {
      return [];
    }
    const data = JSON.parse(fs.readFileSync(STUDENTS_FILE, "utf8"));
    return Array.isArray(data) ? data.map((item) => Student.fromJson(item)) : [];
  }

  static saveGrades(grades) {
    fs.writeFileSync(GRADES_FILE, JSON.stringify(grades.map((grade) => grade.toJson()), null, 2), "utf8");
  }

  static loadGrades() {
    if (!fs.existsSync(GRADES_FILE)) {
      return [];
    }
    const data = JSON.parse(fs.readFileSync(GRADES_FILE, "utf8"));
    return Array.isArray(data) ? data.map((item) => Grade.fromJson(item)) : [];
  }
}

module.exports = JsonManager;

File: ws/collahuazo/u2/ws13GradesStudentAppCLI/Python/controller/json_manager.py
import json
from pathlib import Path

from model.grade import Grade
from model.student import Student

ROOT_PATH = Path(__file__).resolve().parents[1]
STUDENTS_FILE = ROOT_PATH / "students.json"
GRADES_FILE = ROOT_PATH / "grades.json"


class JsonManager:
    @staticmethod
    def save_students(students):
        with open(STUDENTS_FILE, "w", encoding="utf-8") as file:
            json.dump([student.to_dict() for student in students], file, indent=4, ensure_ascii=False)

    @staticmethod
    def load_students():
        if not STUDENTS_FILE.exists():
            return []
        with open(STUDENTS_FILE, "r", encoding="utf-8") as file:
            data = json.load(file)
            return [Student.from_dict(item) for item in data if isinstance(item, dict)]

    @staticmethod
    def save_grades(grades):
        with open(GRADES_FILE, "w", encoding="utf-8") as file:
            json.dump([grade.to_dict() for grade in grades], file, indent=4, ensure_ascii=False)

    @staticmethod
    def load_grades():
        if not GRADES_FILE.exists():
            return []
        with open(GRADES_FILE, "r", encoding="utf-8") as file:
            data = json.load(file)
            return [Grade.from_dict(item) for item in data if isinstance(item, dict)]
